metabase/metabase
# Implement cumulative aggregators in mongo

Closes https://github.com/metabase/metabase/issues/50832

### Description

Previously, cumulative aggregators were being done in clojure after the mongo query was finished.  However, this didn't work on multi-step queries where the cumulative aggregator wasn't the last step (for example, do a cumulative sum and then filter the result).  Implementing cumulative operators directly in mongo fixes this issue.

### How to verify

- Use a mongo db that is version 5 or later
- Create a question using cumulative sum or cumulative count
- Verify that the native query is using a $setWindowFields stage
- Add a filter stage after the cumulative sum/count
- Verify that the output is correct

### Checklist
